Shakemap map

dslosky-usgs opened the pull request:
Closes #925 closes #926 

Initial shakemap map working with InteractiveMapOverlaysPipe and queryParams to display the correct layers.

jmfee-usgs commented on src/app/shakemap/map/map.component.html:
```diff
@@ -0,0 +1,18 @@
+<ng-container *ngIf="eventService?.event$ | async; let event">
+  <ng-container *ngIf="event | getProduct:'shakemap' as shakemap">
+    <ng-container *ngIf="activatedRoute.queryParamMap | async; let params">
```
Shakemap doesn't need to do this.

you should only have:
```
<ng-container *ngIf="eventService.product$ | async; let product">
  ...
  <shared-map
      [overlays]="product | shakemapOverlays:'mmi'"
  ...
```

jmfee-usgs commented on src/app/shakemap/shakemap/shakemap.component.html:
```diff
@@ -8,6 +8,15 @@ <h3>ShakeMap</h3>
+        queryParamsHandling="preserve"
+        routerLinkActive #map="routerLinkActive"
+        [queryParams]="{
+          'shakemap-intensity': true
```
you should not pass this parameter to the shakemap map; only to the interactive map when linking from the static shakemap map component.

jmfee-usgs commented on src/app/shared/map/map.component.ts:
```diff
@@ -177,6 +177,13 @@ export class MapComponent implements AfterViewInit, OnInit {
     return this._showScaleControl;
   }
 
+  fitBounds(bounds) {
```
this should not be a separate method, do this stuff in a setTimeout at the end of the setBounds method.

dslosky-usgs commented:
Simplified ShakemapOverlaysPipe; optionally pass in an overlay ID to enable it. No longer using logic from InteractiveMapOverlaysPipe

Removed ShakemapMapComponent; using the shared-map instead
